urllib3/connectionpool.py

```diff
@@ -568,25 +568,22 @@ def urlopen(self, method, url, body=None, headers=None, retries=None,
             # Close the connection. If a connection is reused on which there
             # was a Certificate error, the next request will certainly raise
             # another Certificate error.
-            if conn:
-                conn.close()
-                conn = None
+            conn = conn and conn.close()
+            release_conn = True
```

Comment:
This could result in a read error on retry when we're not preloading content. release_conn should not be changed since it's used in subsequent recursive calls to this method. If you want to do this you should use a separate (local) variable. Retries should use the same value of release_conn as supplied to this call.